Assets/FullSerializer/Source/Converters/fsDateConverter.cs

```diff
@@ -27,31 +27,45 @@ public class fsDateConverter : fsConverter {
         public override fsResult TrySerialize(object instance, out fsData serialized, Type storageType) {
             if (instance is DateTime) {
                 var dateTime = (DateTime)instance;
-                serialized = new fsData(dateTime.ToString(DateTimeFormatString));
+                if(Serializer.Config.SerializeDateTimeAsInteger) {
+                    serialized = new fsData(dateTime.Ticks);
+                } else {
+                    serialized = new fsData(dateTime.ToString(DateTimeFormatString));
+                    return fsResult.Success;
+                }
                 return fsResult.Success;
             }
 
             if (instance is DateTimeOffset) {
                 var dateTimeOffset = (DateTimeOffset)instance;
-                serialized = new fsData(dateTimeOffset.ToString(DateTimeOffsetFormatString));
+                serialized = new fsData(dateTimeOffset.Ticks);
                 return fsResult.Success;
             }
 
             if (instance is TimeSpan) {
                 var timeSpan = (TimeSpan)instance;
-                serialized = new fsData(timeSpan.ToString());
+                if(Serializer.Config.SerializeDateTimeAsInteger) {
+                    serialized = new fsData(timeSpan.Ticks);
+                } else {
+                    serialized = new fsData(timeSpan.ToString());
+                }
                 return fsResult.Success;
             }
 
             throw new InvalidOperationException("FullSerializer Internal Error -- Unexpected serialization type");
         }
 
         public override fsResult TryDeserialize(fsData data, ref object instance, Type storageType) {
-            if (data.IsString == false) {
```

Comment:
I would drop all of the config checks in this function; deserialization should be successful regardless of the current config mode.